src/core/ext/transport/chttp2/client/secure/secure_channel_create.c

```diff
@@ -61,41 +61,67 @@ static grpc_subchannel_args *get_secure_naming_subchannel_args(
     grpc_exec_ctx *exec_ctx, const grpc_subchannel_args *args) {
   grpc_channel_credentials *channel_credentials =
       grpc_channel_credentials_find_in_args(args->args);
-  // Make sure security connector does not already exist in args.
-  if (grpc_security_connector_find_in_args(args->args) != NULL) {
-    gpr_log(GPR_ERROR, "Security connector already present in channel args.");
-    return NULL;
-  }
   if (channel_credentials == NULL) {
     gpr_log(GPR_ERROR,
             "Can't create subchannel: channel credentials missing for secure "
             "channel.");
     return NULL;
   }
+  // Make sure security connector does not already exist in args.
+  if (grpc_security_connector_find_in_args(args->args) != NULL) {
+    gpr_log(GPR_ERROR,
+            "Can't create subchannel: security connector already present in "
+            "channel args.");
+    return NULL;
+  }
   const grpc_slice_hash_table *targets_info =
       grpc_glb_targets_info_find_in_args(args->args);
-  // To which address are we connecting?
-  const char *target_uri_str = grpc_get_subchannel_address_uri_arg(args->args);
-  grpc_uri *target_uri =
-      grpc_uri_parse(target_uri_str, true /* suppress errors */);
+  // To which address are we connecting? By default, use the server URI.
+  const grpc_arg *server_uri_arg =
+      grpc_channel_args_find(args->args, GRPC_ARG_SERVER_URI);
+  GPR_ASSERT(server_uri_arg != NULL);
+  GPR_ASSERT(server_uri_arg->type == GRPC_ARG_STRING);
+  const char *server_uri_str = server_uri_arg->value.string;
+  GPR_ASSERT(server_uri_str != NULL);
+  grpc_uri *server_uri =
+      grpc_uri_parse(server_uri_str, true /* supress errors */);
+  if (server_uri == NULL) {
```

Comment:
Once you merge in the changes from #9539, you can probably change this to an assertion.